### PR TITLE
Add nightly CI on Frontier

### PR DESCRIPTION
**Pull request by Rombur**
This PR adds a `.gitlab-ci.yml` configuration file to run a nightly CI on Frontier. I use a minimal configuration because that's the configuration used to create the Kokkos module on Frontier. I have disable the `DeviceAndThreads` test because the test uses python and there is an issue when we try to load some python module.

### CODE REVIEW

File: .gitlab-ci.yml
Comment (dalg24) [inline .gitlab-ci.yml:1]: Maybe we should use a less generic name and include "olcf" in the file name.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+test:
+  stage: test
+  tags: [frontier, shell]
+  id_tokens:
+    OLCF_ID_TOKEN:
+      aud: https://code.olcf.ornl.gov
+  script:
+    - module load rocm/6.0
+    - mkdir build
+    - cd build
```
Comment (dalg24) [inline .gitlab-ci.yml:10]: Do we need to clean up and delete the build directory?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    - cmake -DCMAKE_CXX_COMPILER=hipcc -DKokkos_ENABLE_HIP=ON -DKokkos_ENABLE_TESTS=ON ..
```
Comment (dalg24) [inline .gitlab-ci.yml:11]: I'd prefer doing `cmake -B build [options...]` and `cmake --build build`

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    - make -j16
```
Comment (crtrott) [inline .gitlab-ci.yml:12]: since I assume this builds on a compute node anyway: would it reduce runtime to go to -j 40 or so?

Comment (dalg24) [reply .gitlab-ci.yml:12]: or just `cmake --build -j` with no number of processes argument

**Review by dalg24 — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
Did you change anything?  We (or at least I) don;t care about the CI for this change set

**Comment by Rombur** [comment]
> Did you change anything?

There was a couple of typos when I switched to `cmake -B` and `cmake --build`. It's working now

**Pull request merged**